bundles/org.openhab.ui.basic/web-src/smarthome.js

```diff
@@ -1337,8 +1339,37 @@
 				icon.addEventListener("load", _t.convertToInlineSVG);
 				icon.addEventListener("error", _t.replaceImageWithNone);
 			}
+
+			if (maxButtonWidth < button.offsetWidth) {
+				maxButtonWidth = button.offsetWidth;
+			}
 		});
 
+		if (_t.buttons.length > 1 && _t.parentNode.classList.contains(o.buttonsMultilineClass)) {
+			_t.minimizeWidth = function() {
+				var labelLength = _t.label.textContent.trim().length;
+				_t.label.style.paddingLeft = labelLength === 0 ? 0 : null; // null reverts it to the css
+				_t.label.style.minWidth = labelLength < 6 ? "min-content" : null;
+
+				// Minimize the width taken by the buttons without adding extra rows.
+				// Start from the maximum width (limited by `mdl-form__buttons-multiline.max-width`),
+				// then shrink it down to the minimum without causing additional wrapping.
+				var buttons = _t.parentNode;
+				var width = buttons.parentElement.offsetWidth + 100; // start wider than max-width allows
+				buttons.style.width = width + "px";
+				var height = buttons.offsetHeight;
+				while (buttons.offsetHeight === height && width >= maxButtonWidth) {
+					buttons.style.width = --width + "px";
+				}
+				buttons.style.width = (width+1) + "px";
```

Comment:
Maybe we could have a variable and set "buttons.style.width" only at end ?

Reply:
No, that's not possible. We need to _actually_ set and decrease the width incrementally, and we want this to be applied to the actual element, incrementally, so that when it's too narrow, the actual offset height would change, which indicates that the buttons have now wrapped around a new row. That's why we then back off and re-add the width to prevent the wrapping. This can only be done like this that I know of to figure out the breakpoint / threshold.